hashpage.go

```diff
@@ -246,3 +249,98 @@ func (es *ExploreServer) contractPage(w http.ResponseWriter, fcJSON []byte, fcid
 	}
 	w.Write(page)
 }
+
+// findOutput scans a transaction/bolck and returns the output
+func (es *ExploreServer) findOutput(oID types.SiacoinOutputID) (types.SiacoinOutput, error) {
+	// First do a lookup on the txid
+	itemJSON, err := es.apiGetHash(oID[:])
+	if err != nil {
+		return types.SiacoinOutput{}, err
+	}
+
+	var or modules.OutputResponse
+	err = json.Unmarshal(itemJSON, &or)
+	if err != nil {
+		return types.SiacoinOutput{}, err
+	}
+
+	// Will have to switch on the type, as it colud be a block
+	itemJSON, err = es.apiGetHash(or.OutputTx[:])
+	if err != nil {
+		return types.SiacoinOutput{}, err
+	}
+
+	var rd responseData
+	err = json.Unmarshal(itemJSON, &rd)
+	if err != nil {
+		return types.SiacoinOutput{}, err
+	}
+
+	switch rd.ResponseType {
+	case "Block":
+		var b modules.BlockResponse
+		err := json.Unmarshal(itemJSON, &b)
+		if err != nil {
+			return types.SiacoinOutput{}, nil
+		}
```

Comment:
I assume you don't actually want to return `nil` here.

For a function like this, it may be better style to used named return values. Then you can convert all of the return statements to bare `return`s, except `return findOutputBlock` and `return findOutputTransaction`.